pkg/dmaasbackup/schedule.go

```diff
@@ -77,12 +83,159 @@ func (d *dmaasBackup) generateBackupName(dbkp v1alpha1.DMaaSBackup) string {
 // updateScheduleInfo checks for relevant velero schedule/backup and
 // update the dbkp with schedule/backup details
 func (d *dmaasBackup) updateScheduleInfo(dbkp *v1alpha1.DMaaSBackup) error {
-	// TODO
+	// check if dbkp is having config for non-scheduled backup
+	if yes := isSchedule(*dbkp); !yes {
+		bkp, err := d.backupClient.Get(dbkp.Name, metav1.GetOptions{})
+		if err != nil && !apierrors.IsNotFound(err) {
+			return errors.Wrapf(err, "failed to update scheduleInfo")
+		}
+		if bkp.Name == "" {
```

Comment:
how is this possible? isn''t bkp.Name supposed to be same as 'dbkp.Name'?

Reply:
yes. I'll update it.

Reply:
is this possible if error is IsNotFound?

Reply:
if so, better to check for isNotFound error

Reply:
however that need to be handled

Reply:
ok